pkg/promclient/promclient.go

```diff
@@ -88,6 +89,157 @@ func ExternalLabels(ctx context.Context, logger log.Logger, base *url.URL) (labe
+		TSDBMinTime        modelDuration `json:"storage.tsdb.min-block-duration"`
+		TSDBMaxTime        modelDuration `json:"storage.tsdb.max-block-duration"`
+		WebEnableAdminAPI  modelBool     `json:"web.enable-admin-api"`
+		WebEnableLifecycle modelBool     `json:"web.enable-lifecycle"`
+	}{}
+
+	if err := json.Unmarshal(b, &parsableFlags); err != nil {
+		return err
+	}
+
+	*f = Flags{
+		TSDBPath:           parsableFlags.TSDBPath,
+		TSDBRetention:      model.Duration(parsableFlags.TSDBRetention),
+		TSDBMinTime:        model.Duration(parsableFlags.TSDBMinTime),
+		TSDBMaxTime:        model.Duration(parsableFlags.TSDBMaxTime),
+		WebEnableAdminAPI:  bool(parsableFlags.WebEnableAdminAPI),
+		WebEnableLifecycle: bool(parsableFlags.WebEnableLifecycle),
+	}
+	return nil
+}
+
+type modelDuration model.Duration
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (d *modelDuration) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	dur, err := model.ParseDuration(s)
+	if err != nil {
+		return err
+	}
+	*d = modelDuration(dur)
+	return nil
+}
+
+type modelBool bool
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (m *modelBool) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	boolean, err := strconv.ParseBool(s)
+	if err != nil {
+		return err
+	}
+	*m = modelBool(boolean)
+	return nil
+}
+
+// ConfiguredFlags some configured flags from /api/v1/status/flags Prometheus endpoint.
+// Added to Prometheus from v2.2.
+func ConfiguredFlags(ctx context.Context, logger log.Logger, base *url.URL) (Flags, error) {
+	u := *base
+	u.Path = path.Join(u.Path, "/api/v1/status/flags")
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return Flags{}, errors.Wrap(err, "create request")
+	}
+	resp, err := http.DefaultClient.Do(req.WithContext(ctx))
+	if err != nil {
+		return Flags{}, errors.Wrapf(err, "request config against %s", u.String())
+	}
+	defer runutil.CloseWithLogOnErr(logger, resp.Body, "query body")
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return Flags{}, errors.Errorf("failed to read body")
+	}
+
+	if resp.StatusCode != 200 {
+		return Flags{}, errors.Errorf("got non-200 response code: %v, response: %v", resp.StatusCode, string(b))
+	}
+
+	var d struct {
+		Data Flags `json:"data"`
+	}
+	if err := json.Unmarshal(b, &d); err != nil {
+		return Flags{}, errors.Wrapf(err, "unmarshal response: %v", string(b))
+	}
+
+	return d.Data, nil
+}
+
+// Snapshot will request Prometheus to perform snapshot in directory returned by this function.
+// Returned directory is relative to Prometheus data-dir.
+// NOTE: `--web.enable-admin-api` flag has to be set on Prometheus.
+// Added to Prometheus from v2.1.
+func Snapshot(ctx context.Context, logger log.Logger, base *url.URL, skipHead bool) (string, error) {
```

Comment:
would we benefit from any metrics within here? Duration of snapshot request? Total snapshots successful?

Reply:
Also as this is only used in tests I take it that it will be used in the next PR :+1:

Reply:
yup.

Reply:
Added TODO